src/WebJobs.Script/Binding/ExtensionBinding.cs

```diff
@@ -67,13 +70,32 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes(Type para
                 var result = await context.Binder.BindAsync<JObject>(_attributes.ToArray());
                 if (Access == FileAccess.Read)
                 {
-                    context.Value = result;
+                    inputValue = result;
                 }
             }
+            else if (_binding.DefaultType == typeof(JArray))
+            {
+                JArray entityArray = await context.Binder.BindAsync<JArray>(_attributes.ToArray());
+                inputValue = entityArray;
+            }
             else
             {
                 throw new NotSupportedException($"ScriptBinding type {_binding.DefaultType} is not supported");
             }
+
+            if (inputValue != null)
+            {
+                if (context.DataType == DataType.Stream)
```

Comment:
The existing table binding only did this conversion when the Access was not Write (i.e. only on reads)